internal/status/cache.go

```diff
@@ -87,3 +106,45 @@ func (c Cache) getProxyStatusUpdates() []k8s.StatusUpdate {
 	return psu
 
 }
+
+func (c Cache) GetProxyValidConditions() map[types.NamespacedName]projcontour.DetailedCondition {
+	validConds := make(map[types.NamespacedName]projcontour.DetailedCondition)
+
+	for fullname, pu := range c.proxyUpdates {
+		validConds[fullname] = *pu.Conditions[ValidCondition]
+	}
+
+	return validConds
+}
+
+type Metric struct {
+	Vhost     string
+	Status    ProxyStatus
+	Namespace string
+}
+
+func (c Cache) GetProxyStatusMetrics() []Metric {
```

Comment:
I would prefer to avoid having this function. The caller should be able to implement this in terms of `GetProxyValidConditions()`.